hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/codec/FileServiceStateTranslator.java

```diff
@@ -0,0 +1,113 @@
+    }
+
+    /**
+     * The method converts a {@link com.hedera.node.app.service.mono.files.HFileMeta} content and metadata  to a {@link File}
+     * @param data content of the file in old state
+     * @param metadata metadata of the file in old state
+     * @param fileID old protobuf fileID that will be used in order to initialize the new protobuf file
+     * @return new protobuf File
+     */
+    @NonNull
+    public static File stateToPbj(
+            @Nullable final byte[] data,
+            @NonNull final com.hedera.node.app.service.mono.files.HFileMeta metadata,
+            @NonNull final com.hederahashgraph.api.proto.java.FileID fileID) {
+        requireNonNull(metadata);
+        requireNonNull(fileID);
+        final var fileBuilder = new File.Builder();
+        fileBuilder.fileNumber(fileID.getFileNum());
+        fileBuilder.expirationTime(metadata.getExpiry());
+        if (metadata.getWacl() != null)
+            fileBuilder.keys(PbjConverter.asPbjKey(metadata.getWacl()).keyList());
+        if (data != null) fileBuilder.contents(Bytes.wrap(data));
+        fileBuilder.memo(metadata.getMemo());
+        fileBuilder.deleted(metadata.isDeleted());
+
+        return fileBuilder.build();
+    }
+
+    /**
+     * The method converts a {@link File} to a {@link com.hedera.node.app.service.mono.files.HFileMeta} content and metadata
+     * @param fileID new protobuf fileID that will be used to search in file store
+     * @param readableFileStore file store that will be used to search for the file
+     * @return File and metadata pair object
+     */
+    @NonNull
+    public static FileMetadataAndContent pbjToState(
+            @NonNull FileID fileID, @NonNull ReadableFileStoreImpl readableFileStore) throws DecoderException {
+        requireNonNull(fileID);
+        requireNonNull(readableFileStore);
+        final var optionalFile = readableFileStore.getFileLeaf(fileID);
+        return pbjToState(optionalFile.orElseThrow(() -> new IllegalArgumentException("File not found")));
+    }
+
+    /**
+     * The method converts a {@link File} to a {@link FileMetadataAndContent} content and metadata
+     * @param file new protobuf file that will be used
+     * @return File and metadata pair object
+     */
+    @NonNull
+    public static FileMetadataAndContent pbjToState(@NonNull File file) throws DecoderException {
+        requireNonNull(file);
+        var keys = (file.hasKeys())
+                ? com.hedera.node.app.service.mono.legacy.core.jproto.JKey.convertKey(
+                        Key.newBuilder().keyList(file.keys()).build(), 1)
+                : null;
+        com.hedera.node.app.service.mono.files.HFileMeta hFileMeta =
+                new HFileMeta(file.deleted(), keys, file.expirationTime(), file.memo());
+        final byte[] data = (file.contents() == null) ? null : file.contents().toByteArray();
+        return new FileMetadataAndContent(data, hFileMeta);
```

Comment:
FileMetadataAndContent parameters are marked @NonNull, but data could be null

Reply:
FileMetadataAndContent as object cannot be null the properties in it can be

Reply:
Line 106 conditionally sets the value of `data` variable to null
In line 107, we pass `data` to `FileMetadataAndContent` constructor
But according to definition of `FileMetadataAndContent` (line 112), the `data` parameter is @NonNull

Reply:
made it nullable